### DISCUSSION TITLE
Graceful way to hook into the relationship service?

### DISCUSSION THREAD

**Discussion by james-whittington1** [leekelleher/umbraco-contentment#296] [Question]
Hello all, I am investigating how to build a picker that displays related content as a picker.

I think that contentment allows most of what I need, I can set it up so that I can select a particular relationship type and a content type "target" 

The datalist would then display all the content of the target type, and allow it to be selected.

For example datalists that connects staff and articles.

Added to staff document type
Datatype Name: Relate staff to Article
relationshipType: staff_x_article (a bidirectional relationship)
contentType: Article

Added to article document type
Datatype Name: Relate article to staff
relationshipType: staff_x_article  (a bidirectional relationship)
contentType: Staff

These datatypes would releate the current document to the items selected in the picker.

The part that I am not so sure about is how to actually get this property backed by the relation service... I can also see a glaring issue that any "external" manipulation of the relations occur the property editor would not be able handle this.

I have investigated how this could be achived using the notification service, and I _think_ everthing is possible, but I am not sure if this is a good idea, or if it so far out of scope that it should be abandoned :)

**Comment by s6admin** [comment]
Hi james-whittington1,

I'm not the one you were hoping to hear from, but we're looking at similar scenarios where picking targets would result in a (two-way) relation record rather than node ids. Currently stuck in a forked 4.4.0 branch, but unable to get the DataListValueConverter to hit ConvertSourceToIntermediate or ConvertIntermediateToXPath methods which is where I expected to put the relation logic.

Wondering if its something we haven't set up properly or if the Contentment project isn't inheriting PropertyValueConverterBase correctly. Hoping leekelleher might have some insight towards that in a new discussion.

**Reply by leekelleher** [reply]
For posterity, replied on the other thread: https://github.com/leekelleher/umbraco-contentment/discussions/300#discussioncomment-4750905

**Comment by leekelleher** [comment]
@james-whittington1 I hadn't responded previously because I don't have a reasonable suggestion of how Data List and the Relation Service should work together. (I've unsuccessfully attempted to do this previously, ended up with edge cases that stumped me).

If you do (or did) come up with an implementation, I'd be happy to hear it. 😃
